packages/hash/frontend/src/components/block-loader/block-loader.tsx

```diff
@@ -50,53 +47,105 @@ type BlockLoaderProps = {
-      entityId: entityId ?? "entityId-not-yet-set", // @todo ensure blocks always get sent an entityId
-      entityTypeId,
-      properties: entityProperties,
-    });
-
-    return {
-      blockEntity,
-      entityTypes: uniqBy(
-        convertedEntityTypesForProvidedEntities,
-        "entityTypeId",
-      ),
-      readonly: isReadonlyMode,
-      // We currently don't construct a block graph or linked aggregations.
-      blockGraph: {} as BlockGraph,
-      linkedAggregations: [],
+  const [graphProperties, setGraphProperties] = useState<Required<
+    BlockGraphProperties["graph"]
+  > | null>(null);
+
+  const { getEntity } = useBlockProtocolGetEntity();
+
+  useEffect(() => {
+    const depths: GraphResolveDepths = {
+      hasRightEntity: {
+        incoming: 2,
+        outgoing: 2,
+      },
+      hasLeftEntity: {
+        incoming: 2,
+        outgoing: 2,
+      },
     };
-  }, [entityId, entityProperties, entityTypeId, isReadonlyMode]);
+
+    if (!blockEntityId) {
+      // when inserting a block in the frontend we don't yet have an entity associated with it
+      // there's a delay while the request to the API to insert it is processed
+      // @todo some better way of handling this – probably affected by revamped collab.
+      //    or could simply not load a new block until the entity is created?
+      const now: string = new Date().toISOString();
+      const placeholderEntity = {
+        metadata: {
+          editionId: {
+            baseId: "placeholder-account%entity-id-not-set",
+            version: now, // @todo-0.3 check this against types in @blockprotocol/graph when mismatches fixed
+            versionId: now,
+          },
+          entityTypeId: blockEntityTypeId,
+        },
+
+        properties: {},
+      };
+      const blockEntitySubgraph = {
+        depths,
+        edges: {},
+        roots: [placeholderEntity.metadata.editionId as any as EntityEditionId], // @todo-0.3 fix when type mismatches fixed
+        vertices: {
+          [placeholderEntity.metadata.editionId.baseId]: {
+            [now]: {
+              kind: "entity",
+              inner: placeholderEntity,
+            },
+          },
+        } as unknown as Subgraph["vertices"], // @todo-0.3 do something about this
+      };
+      setGraphProperties({
+        blockEntitySubgraph,
+        readonly: isReadonlyMode,
+      });
+      return;
+    }
+
+    getEntity({
+      data: { entityId: blockEntityId },
+    })
+      .then(({ data, errors }) => {
+        if (!data) {
+          throw new Error(
+            `Could not get entity ${blockEntityId} ${
+              errors ? JSON.stringify(errors, null, 2) : ""
+            }`,
+          );
+        }
+
+        setGraphProperties({
+          blockEntitySubgraph: {
+            ...(data as unknown as Subgraph<SubgraphRootTypes["entity"]>), // @todo-0.3 do something about this,
```

Comment:
I appreciate where these casts have come from, I've got something similar in my branch but I'm not sure we should commit it to `main`, even temporarily. Just feels like there are far far too many gotchas and we may end up with some non-insignificant lost dev time to bugs we accidentally introduce here.

What do you think about blocking this PR on us either updating the BP package, or introducing a refactoring PR in HASH to address the mismatches?

Reply:
We spoke about this and agreed to get this PR in on the basis that I would fix the type mismatches soon this week (following a discussion on approach tomorrow), and warn anyone who was planning to work in this area about possibly waiting until they are fixed